lib/metadataTypes/MetadataType.js

```diff
@@ -628,14 +629,15 @@ class MetadataType {
      * @param {string} uri rest endpoint for GET
      * @param {string} [overrideType] force a metadata type (mainly used for Folders)
      * @param {TYPE.TemplateMap} [templateVariables] variables to be replaced in the metadata
+     * @param {string|number} [singleRetrieve] key of single item to filter by
      * @returns {Promise.<{metadata: (TYPE.MetadataTypeMap | TYPE.MetadataTypeItem), type: string}>} Promise of item map (single item for templated result)
      */
-    static async retrieveREST(retrieveDir, uri, overrideType, templateVariables) {
-        const response = this.definition.restPagination
-            ? await this.client.rest.getBulk(uri)
-            : await this.client.rest.get(uri);
-        const results = this.parseResponseBody(response);
-
+    static async retrieveREST(retrieveDir, uri, overrideType, templateVariables, singleRetrieve) {
```

Comment:
can we consider a different method for this? it feels like a little overloading of params

Reply:
hm. im thinking given that it's supposed to be the one generic "retrieveREST" method used by all it's bound to become a bit messy.